### PR TITLE
Refactor and prep for CI/CD pipeline

### PR DESCRIPTION
**Pull request by thomasrayner**
Pursuant to #1, I've done a small refactor on the code - just moving the `using` statements around, and added a couple scripts to support the automated deployment of a single `.psm1` instead of a group of `.ps1` files. 

You'll note that using statements now have to go in the `PSWordCloud\Private\psm1include.ps1` file to be included in the final `.psm1` in the right spot. I've hard coded the path to that file for now, but it could easily be parameterized or turned into an array in the future.

I've also added a `.gitignore` file to prevent accidental committing of build artifacts.

These scripts are setup so that you can run them locally or in Azure DevOps and have the same experience. `\Build\refactor.ps1` will output the "built" module. The PSSA files are just for running PSSA in Azure DevOps. Locally, you'd probably just run `Invoke-ScriptAnalyzer` instead.

### CODE REVIEW

File: Build/runpssa.ps1 [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+process {
+    $moduleBase = Split-Path $PSD1Path -Parent
+    $moduleName = (Split-Path $PSD1Path -Leaf) -replace '.psd1$'
+
+
+    $testFile = New-TemporaryFile
+    Rename-Item -Path $testFile -NewName "$($testFile.BaseName).ps1"
+    $testFilePS1 = "$($testFile.Directory)\$($testFile.basename).ps1"
+
+    $testFileContent = @"
```
Comment (vexx32) [inline Build/runpssa.ps1]: Use a literal here-string `@'` instead; will save us a _lot_ of pesky backticks here. 😄

File: Build/refactor.ps1 [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        $psm1Content | Add-Content -Path "$outputDir\$($moduleName).psm1" 
+
+    }
+    catch {
+        throw "Unable to combine PS1 files into PSM1"
+    }
+
+    try {
+        Write-Verbose "Updating FunctionsToExport property in output PSD1"
+        Update-ModuleManifest -Path  "$outputDir\$moduleName.psd1" -FunctionsToExport $public.BaseName
```
Comment (vexx32) [inline Build/refactor.ps1]: Can we use this to update aliases too, somehow? I tend to declare them in `[Alias()]` attributes...

File: Build/refactor.ps1 [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        try {
+            $null = Copy-Item -Path $CopyDirectories -Recurse -Destination $outputDir -Force
+        }
+        catch {
+            throw "Unable to gather copy directories"
+        }
+    }
+
+    try {
+        Write-Verbose "Contactenating public and private functions into PSM1 in output dir"
```
Comment (vexx32) [inline Build/refactor.ps1]:
```suggestion
        Write-Verbose "Concatenating public and private functions into PSM1 in output directory"
```

File: Build/refactor.ps1 [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+)
+
+begin {
+    $eap = $ErrorActionPreference
+    $ErrorActionPreference = 'Stop'     # Make errors terminate instead of fighting through them
+}
+
+process {
+    $moduleBase = Split-Path $PSD1Path -Parent
+    $moduleName = (Split-Path $PSD1Path -Leaf) -replace '.psd1$'
```
Comment (vexx32) [inline Build/refactor.ps1]:
```suggestion
    $moduleName = (Get-Item -Path $PSD1Path).Basename
```

**Comment by thomasrayner** [comment]
> Couple questions here, and this is pretty solid!
> 
> You did test the module actually builds with this, right? 😉
> 
> I'll check it tomorrow lol

Haha, yeah I tested the module it builds. Seems to work, as far as I can tell. Will take a look at your suggestions asap. Some look easy and obvious, others less so. Hopefully there's an easy way to gather those aliases.

**Comment by vexx32** [comment]
Closing for now as I've go the pipeline mostly taken care of. There are some pieces in here that I'd love to incorporate, so if you're game to add the missing pieces from this into my current implementation I'm more than happy to review a more up to date PR if you'd like to submit another. 

Thanks for your work on this one! 💖 😊 🌸

**Pull request closed without merging**